src/v/cluster/self_test_rpc_types.h

```diff
@@ -289,6 +289,27 @@ struct self_test_result
           r.error ? *r.error : "<no_value>");
         return o;
     }
+
+    auto serde_fields() {
+        return std::tie(
+          p50,
+          p90,
+          p99,
+          p999,
+          max,
+          rps,
+          bps,
+          timeouts,
+          test_id,
+          name,
+          info,
+          test_type,
+          duration,
+          warning,
+          error,
+          start_time,
+          end_time);
+    }
```

Comment:
nice. i kinda think we should change serde to require using serde_fields instead of the implicit ordering in the struct?

Reply:
Yeah agreed. I don't really get the need for it generally; besides which i believe it maxes out at 20 fields or something like that. Pretty marginal as a convenience feature IMO.

Reply:
I'd be happy to knock that out soon. See what people think